cdap-app-fabric/src/test/java/co/cask/cdap/gateway/handlers/InMemoryAuthorizer.java

```diff
@@ -46,22 +61,96 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
 
   @Override
   public void grant(EntityId entity, Principal principal, Set<Action> actions) {
-    get(entity, principal).addAll(actions);
+    getActions(entity, principal).addAll(actions);
   }
 
   @Override
   public void revoke(EntityId entity, Principal principal, Set<Action> actions) {
-    get(entity, principal).removeAll(actions);
+    getActions(entity, principal).removeAll(actions);
   }
 
   @Override
   public void revoke(EntityId entity) {
     for (Principal principal : table.row(entity).keySet()) {
-      get(entity, principal).clear();
+      getActions(entity, principal).clear();
+    }
+  }
+
+  @Override
+  public void createRole(Role role) throws RoleAlreadyExistsException {
+    if (roleToPrincipalMap.containsKey(role)) {
+      throw new RoleAlreadyExistsException(role);
+    }
+    roleToPrincipalMap.put(role, new HashSet<Principal>());
+  }
+
+  @Override
+  public void dropRole(Role role) throws RoleNotFoundException {
```

Comment:
I think the contract of this method (not just in-memory) should also include that a role cannot be dropped, if some principals have it assigned still. So,
```
if (roleToPrincipalMap.containsKey(role) && !roleToPrincipalMap.get(role).isEmpty()) {
  throw CannotBeDeletedException(role);
}
```